src/main/java/org/opensearch/knn/index/query/filtered/FilteredIdsKNNByteIterator.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.query.filtered;
+
+import org.apache.lucene.index.BinaryDocValues;
+import org.apache.lucene.search.DocIdSetIterator;
+import org.apache.lucene.util.BitSet;
+import org.apache.lucene.util.BitSetIterator;
+import org.apache.lucene.util.BytesRef;
+import org.opensearch.knn.index.SpaceType;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+
+/**
+ * Inspired by DiversifyingChildrenFloatKnnVectorQuery in lucene
+ * https://github.com/apache/lucene/blob/7b8aece125aabff2823626d5b939abf4747f63a7/lucene/join/src/java/org/apache/lucene/search/join/DiversifyingChildrenFloatKnnVectorQuery.java#L162
+ *
+ * The class is used in KNNWeight to score filtered KNN field by iterating filterIdsArray.
+ */
+public class FilteredIdsKNNByteIterator implements KNNIterator {
```

Comment:
you don't need another class here. You can just pass the vector dataType to the iterator and then based on that you can easily find out the what the byte array should be converted to.

Also adding just a <T> in the FilterIds iterator will do the trick for computing the scores using space type.

Reply:
Separate class is more cleaner. Otherwise you will need if/else statement here and there. Also, the query type is different between the two. `float[]` and `byte[]`.

Reply:
There is unnecessary interfaces which are added and just making this simple iteration logic duplicated across classes. Better idea might be to use <T> here.